src/DataGrid.tsx

```diff
@@ -215,96 +267,216 @@ function DataGrid<R, K extends keyof R, SR>({
+        '--filter-row-height': `${headerFiltersHeight}px`,
+        '--row-width': `${totalColumnWidth}px`,
+        '--row-height': `${rowHeight}px`
+      } as React.CSSProperties}
       ref={gridRef}
+      onScroll={onScroll}
     >
-      {columnMetrics && (
+      <HeaderRow<R, K, SR>
+        rowKey={rowKey}
+        rows={rows}
+        columns={viewportColumns}
+        onColumnResize={handleColumnResize}
+        lastFrozenColumnIndex={lastFrozenColumnIndex}
+        draggableHeaderCell={props.draggableHeaderCell}
+        onHeaderDrop={props.onHeaderDrop}
+        allRowsSelected={selectedRows?.size === rows.length}
+        onSelectedRowsChange={onSelectedRowsChange}
+        sortColumn={props.sortColumn}
+        sortDirection={props.sortDirection}
+        onSort={props.onSort}
+      />
+      {enableFilters && (
+        <FilterRow<R, SR>
+          lastFrozenColumnIndex={lastFrozenColumnIndex}
+          columns={viewportColumns}
+          filters={props.filters}
+          onFiltersChange={props.onFiltersChange}
+        />
+      )}
+      {rows.length === 0 && props.emptyRowsView ? createElement(props.emptyRowsView) : (
         <>
-          <div
-            ref={headerRef}
-            className="rdg-header"
-          >
-            <HeaderRow<R, K, SR>
-              rowKey={rowKey}
-              rows={rows}
-              height={headerRowHeight}
-              width={columnMetrics.totalColumnWidth + getScrollbarSize()}
-              columns={viewportColumns}
-              onColumnResize={handleColumnResize}
-              lastFrozenColumnIndex={columnMetrics.lastFrozenColumnIndex}
-              draggableHeaderCell={props.draggableHeaderCell}
-              onHeaderDrop={props.onHeaderDrop}
-              allRowsSelected={selectedRows?.size === rows.length}
-              onSelectedRowsChange={onSelectedRowsChange}
-              sortColumn={props.sortColumn}
-              sortDirection={props.sortDirection}
-              onSort={props.onSort}
-            />
-            {enableFilters && (
-              <FilterRow<R, SR>
-                height={headerFiltersHeight}
-                width={columnMetrics.totalColumnWidth + getScrollbarSize()}
-                lastFrozenColumnIndex={columnMetrics.lastFrozenColumnIndex}
-                columns={viewportColumns}
-                filters={props.filters}
-                onFiltersChange={props.onFiltersChange}
-              />
-            )}
-          </div>
-          {rows.length === 0 && props.emptyRowsView ? createElement(props.emptyRowsView) : (
-            <Canvas<R, K, SR>
-              ref={ref}
-              rowKey={rowKey}
-              rowHeight={rowHeight}
-              rowRenderer={props.rowRenderer}
-              rows={rows}
-              selectedRows={selectedRows}
-              onSelectedRowsChange={onSelectedRowsChange}
-              columnMetrics={columnMetrics}
+          <InteractionMasks<R, SR>
```

Comment:
I noticed the scrollbar issue is fixed when we use `viewportWidth > 0 && InteractionMasks<R, SR>`

Reply:
https://github.com/adazzle/react-data-grid/blob/allsticky/src/masks/InteractionMasks.tsx#L90
It's caused by this one line.
We should not be stealing focus on page/grid load anyway, I think we should revisit this.